src/Diagnostics.DataProviders/ConfigurationFactory.cs

```diff
@@ -32,15 +32,19 @@ public AppSettingsDataProviderConfigurationFactory(IHostingEnvironment env)
 
             var builtConfig = builder.Build();
 
-            var tokenProvider = new AzureServiceTokenProvider(azureAdInstance: builtConfig["Secrets:AzureAdInstance"]);
-            var keyVaultClient = new KeyVaultClient(
-                new KeyVaultClient.AuthenticationCallback(
-                    tokenProvider.KeyVaultTokenCallback
-                )
-            );
-
-            string keyVaultConfig = Helpers.GetKeyvaultforEnvironment(env.EnvironmentName);
-            builder.AddAzureKeyVault(builtConfig[keyVaultConfig], keyVaultClient, new DefaultKeyVaultSecretManager());
+            // For production and staging, skip outbound call to keyvault for AppSettings
+            if (env.IsDevelopment())
```

Comment:
Should we also use the same boolean condition that you have in startup code? `builtConfig.GetValue<bool>("Secrets:KeyVaultEnabled", true) || hostingEnvironment.IsDevelopment()`

Reply:
`Secrets:KeyVaultEnabled` - this config is present only in CompilerHost. So thats why I took this approach for runtimehost - If development include keyvault else skip the call.

Reply:
Okay. Just quick clarification for N-Clouds, I am using keyvault app settings reference specifically for observer in N-Clouds. Haven't transitioned to using that for runtimehost. You think we can include the `KeyVaultEnabled` flag as part of your condition? Then I'll just keep it enabled like the old way in N-clouds until I move to using keyvault appsetting references or the new approach we're taking in public.

Reply:
Okay sounds good! Thanks for clarification @hforeste